app/controllers/api/base_controller.rb

```diff
@@ -1,5 +1,54 @@
 module API
-  class BaseController < ApplicationController
+  class BaseController < ActionController::API
     include API::TokenAuthenticatable
+
+    before_action :remove_charset
+    rescue_from ActionController::ParameterMissing, with: :missing_parameter_response
+    rescue_from ActionController::UnpermittedParameters, with: :unpermitted_parameter_response
+    rescue_from ActionController::BadRequest, with: :bad_request_response
+    rescue_from ActiveRecord::StatementInvalid, with: :bad_request_response
+    rescue_from ArgumentError, with: :bad_request_response
+    rescue_from ActiveRecord::RecordNotUnique, with: :bad_request_response
+    rescue_from ActiveRecord::RecordInvalid, with: :invalid_transition
+    rescue_from API::Errors::InvalidTransitionError, with: :invalid_transition
```

Comment:
I wonder if we should only keep the generic error messages and remove the more explicit ones till we get to the implementation, like outcomes etc?

Reply:
I've removed all the custom error rescues 👍